### PR TITLE
Infinum/feature/input field

### PR DESCRIPTION
**Pull request by ZvonimirMedak**
# Description

I've added an InputFieldView and everything that's needed for it to work. I've also added a Combine KeyboardHandler which handles the animations of constraints. Created examples for InputFieldView and all of its states and also an example for the keyboard handler. Wrote some tests for the InputFieldView. 

# Checklist:

- [ x ] Have checked there is no same component already in catalog.
- [ x ] Have created a sufficient example or wrote tests for it.
- [ x ] Code is structured, well written using standard Swift coding style.
- [ x ] All public methods and properties have meaningful and concise documentation.
- [ x ] Used `public` modifier for all method and properties that could be changed from user of your feature, and `private` for internal properties.
- [ x ] Removed any reference to the project that piece of code was created in.
- [ x ] Reduced the dependencies to minimum.

### CODE REVIEW

File: Sources/UI/UIView/UIView+HiddenExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    /// ex. if you set isHidden to false 3 times, you have to set it to true 3 times to show it.
+    ///
+    /// - Note: https://stackoverflow.com/a/45599835/9753141
+    var isHiddenInStackView: Bool {
+        get { isHidden }
+        set { setHiddenInStackView(newValue) }
+    }
+
+    private func setHiddenInStackView(_ hidden: Bool) {
+        if hidden {
```
Comment (dejanskledar) [inline Sources/UI/UIView/UIView+HiddenExtensions.swift:44]:
```swift
hidden ? hide() : show()
```

**Review by dejanskledar — APPROVED** [review]
Nicely done, some very good magic in here! Will be very useful :) 🎉

File: Sources/UI/InputField/Base/BaseInputFieldView.swift [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    }
+}
+
+// MARK: - Editig state
+
+@available(iOS 13.0, *)
+public extension BaseInputFieldView {
+
+    func setupEditingState(isEditing: Bool) {
+        let color = errorText.isBlank ? .white : errorColor
```
[the remark this answers is not in the document]
Comment (ZvonimirMedak) [reply Sources/UI/InputField/Base/BaseInputFieldView.swift:109]: For now, it doesn't support both themes 😞

**Comment by nikolamajcen** [comment]
@ZvonimirMedak I've seen this has been approved, so I believe it is ready to be merged?

**Pull request merged**